### PR TITLE
fix inconsistent variable in yield and from clause

### PR DESCRIPTION
**Pull request by czpmango**
#### What type of PR is this?
- [x] bug
- [ ] feature
- [ ] enhancement

#### What does this PR do?
Fix inconsistent variable in yield and from clause.

#### Which issue(s)/PR(s) this PR relates to?
[https://github.com/vesoft-inc/nebula/issues/3210](https://github.com/vesoft-inc/nebula/issues/3210)

#### Checklist：
- [x] Documentation affected （Please add the label if documentation needs to be modified.)
- [ ] Incompatible （If it is incompatible, please describe it and add corresponding label.）
- [x] Need to cherry-pick （If need to cherry-pick to some branches, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

#### Release notes：
                       `

### CODE REVIEW

**Review by Shylock-Hg — APPROVED** [review]

File: src/graph/validator/GoValidator.cpp [hunk 1]
```diff
@@ -36,9 +36,23 @@ Status GoValidator::validateImpl() {
...
+    }
+    auto varPropsMap = exprProps.varProps();
+    std::vector<std::string> keys;
+    for (const auto& [k, v] : varPropsMap) {
+      keys.emplace_back(k);
+    }
+    if (keys.size() > 1) {
+      return Status::SemanticError("Multiple variable property is not supported in WHERE or YIELD");
+    }
+    if (keys.front() != goCtx_->from.userDefinedVarName) {
```
Comment (nevermore3) [inline src/graph/validator/GoValidator.cpp:52]: This condition may never be reached

Comment (czpmango) [reply src/graph/validator/GoValidator.cpp:52]: Why?

Comment (nevermore3) [reply src/graph/validator/GoValidator.cpp:52]: An error will be reported in the validatorFrom. when keys.front() != userdefinedVarName,

Comment (czpmango) [reply src/graph/validator/GoValidator.cpp:52]: I can debug this piece of code use this statement:
```
      $A = GO FROM 'Tim Duncan' OVER like YIELD like._dst AS dst;
       $B = GO FROM $A.dst OVER like YIELD like._dst AS dst;
        GO FROM $A.dst over like YIELD like._dst AS dst, $B.dst
```

Comment (nevermore3) [reply src/graph/validator/GoValidator.cpp:52]: my fault, deduceProp do not work on from clause

**Review by nevermore3 — APPROVED** [review]

**Pull request merged**